### PR TITLE
Fix return value when using convert_windows_string

### PR DESCRIPTION
**Pull request by cristgl**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/3341|

## Description

To fix this bug it was necessary to check if the returned value was NULL.
I have added few more changes to the code:
- Removed some functions that were not being called in the project.
- Added the specific check to avoid the bug.
- Added some additional information to error messages.
    · The `get_message` function returns the error code converted to string and the provider name from which the message could not be accessed.
- Changed the `if` statement that checks the `get_message` function. Now it checks either if the provider name is NULL or the `get_message` has succeeded. This has been made to make the statement more clear as the fact that the provider name is not null comes inherent in the `get_message` call, where this field is used to retrieve the event's message.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
- [x] Source installation
- [x] Source upgrade
- Memory tests
  - [x] Valgrind report for affected components

### CODE REVIEW

File: src/logcollector/read_win_event_channel.c [hunk 1]
```diff
@@ -487,64 +496,65 @@ void send_channel_event(EVT_HANDLE evt, os_channel *channel)
...
-        if(beg_prov){
-            end_prov = strchr(beg_prov+1, '\'');
-
-            if (end_prov){
-                num = end_prov - beg_prov - 1;
-
-                if(num > OS_MAXSTR - 1){
-                    mwarn("The event message has exceeded the maximum size.");
-                    goto cleanup;
+    if (xml_event) {
```
Comment (chemamartinez) [inline src/logcollector/read_win_event_channel.c]: Instead of encapsulating the entire block of code in an `if` statement, include the next check before this line:

```
if (!xml_event) {
    goto clean_up;
}
```

It is equivalent and more clear.

**Comment by chemamartinez** [comment]
More changes:

**Comment by chemamartinez** [comment]
More changes related to the use of `convert_windows_string()` in the rest of the project:

- It is used by `get_property_value()`:
https://github.com/wazuh/wazuh/blob/12484f02f419294962809fbacd58c161129eaf61/src/logcollector/read_win_event_channel.c#L132
But that function is not used in the project.

- In Whodata, it affects to `user_name` and `process_name`:
https://github.com/wazuh/wazuh/blob/12484f02f419294962809fbacd58c161129eaf61/src/syscheckd/win_whodata.c#L527
https://github.com/wazuh/wazuh/blob/12484f02f419294962809fbacd58c161129eaf61/src/syscheckd/win_whodata.c#L534
And they can be NULL when freeing them:
https://github.com/wazuh/wazuh/blob/12484f02f419294962809fbacd58c161129eaf61/src/syscheckd/win_whodata.c#L849
Change that line by `os_free(user_name);` and the same for `process_name`.

**Comment by cristgl** [comment]
`get_property_value()` is not used in the project anymore, as it was called at the `send_channel_event` function after calling `EvtRender` with the flag `EvtRenderEventValues` to get one by one some of the values included in the `PEVT_VARIANT` variable filled in the function.
It has been replaced by `EvtRenderEventXml` to get the event in XML format.

File: src/logcollector/read_win_event_channel.c [hunk 2]
```diff
@@ -240,10 +132,19 @@ char *get_message(EVT_HANDLE evt, LPCWSTR provider_name, DWORD flags)
                                          0,
                                          0);
     if (publisher == NULL) {
+        LSTATUS err = GetLastError();
```
Comment (chemamartinez) [inline src/logcollector/read_win_event_channel.c]: What is the motivation for this change? Could you describe it and show a log sample?

File: src/logcollector/read_win_event_channel.c [hunk 3]
```diff
@@ -510,14 +351,14 @@ void send_channel_event(EVT_HANDLE evt, os_channel *channel)
             }
         }
     }
-    
+
     if (provider_name) {
         wprovider_name = convert_unix_string(provider_name);
 
-        if (wprovider_name && (msg_from_prov = get_message(evt, wprovider_name, EvtFormatMessageEvent)) == NULL) {
+        if (wprovider_name == NULL || (msg_from_prov = get_message(evt, wprovider_name, EvtFormatMessageEvent)) == NULL) {
```
Comment (chemamartinez) [inline src/logcollector/read_win_event_channel.c]: Why are we changing the logic here?

File: src/logcollector/read_win_event_channel.c (hunk 2 above)
Comment (cristgl) [reply src/logcollector/read_win_event_channel.c]: As the message is obtained from the provider name included in the event, it could be useful to show which is the value for this field, as well as converting to string the error code returned. Here is an example:

image (5)

File: src/logcollector/read_win_event_channel.c (hunk 3 above)
Comment (cristgl) [reply src/logcollector/read_win_event_channel.c]: The `get_message` function gets the message from the provider name, which means that if it does not exist, then an error should be returned. It is not necessary to check if the `wprovider_name` exists, as it is inherent to the `get_message` call, but I think it is more clear with this check.

Comment (chemamartinez) [reply src/logcollector/read_win_event_channel.c]: I think we should separate this logic to show two different error messages. It is not the same to have an empty provider name and to not be able to obtain the message. And the same error is shown in both cases (`Could not get message for ...`).

File: src/logcollector/read_win_event_channel.c (hunk 2 above)
Comment (chemamartinez) [reply src/logcollector/read_win_event_channel.c]: Ok, I agree with you that now is more descriptive. However, its level should be changed to `debug1` due to we have another error message more user-friendly when this error happens. It would look like:

```
2019/05/28 11:08:44 ossec-agent[928] read_win_event_channel.c:246 at get_message(): DEBUG: Could not EvtOpenPublisherMetadata() with flags (1) which returned (1717): The interface is unknown. 
2019/05/28 11:08:44 ossec-agent[928] read_win_event_channel.c:522 at send_channel_event(): ERROR: Could not get message for (System), provider (Microsoft-Windows-WinRM)
```

The second error would be the one shown by default which is less technical for the user.

File: src/logcollector/read_win_event_channel.c (hunk 3 above)
Comment (cristgl) [reply src/logcollector/read_win_event_channel.c]: Updated [b1b0edf](https://github.com/wazuh/wazuh/pull/3393/commits/b1b0edfd12f0b63ae59e5563dfd18be2eb049663)

File: src/logcollector/read_win_event_channel.c (hunk 2 above)
Comment (cristgl) [reply src/logcollector/read_win_event_channel.c]: Updated [b1b0edf](https://github.com/wazuh/wazuh/pull/3393/commits/b1b0edfd12f0b63ae59e5563dfd18be2eb049663)
